File: app/schemas/document.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from enum import Enum


class DocumentResponse(BaseModel):
    id: uuid.UUID
    filename: str
    document_type: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True
        

class DocumentType(str, Enum):
    invoice = "invoice"
    resume = "resume"
    contract = "contract"
    research_paper = "research_paper"
    general = "general"

File: frontend/types/document.ts
import { UUID } from "crypto";

export type DocumentType =
    | "invoice"
    | "resume"
    | "contract"
    | "research_paper"
    | "general";

export interface DocumentResponse {
    id: string;
    filename: string;
    document_type: DocumentType;
    status: string;
    created_at: string;
}

export interface BatchUploadResult {
    filename: string;
    success: boolean;
    document_id: string | null;
    error: string | null;
}